### PR TITLE
KF5 additional packages

### PR DESCRIPTION
**Pull request by dutsky**
additional KDE Framework 5 packages(along with previous PRs), needed to build KDE Plasma and KDE Applications.

### CODE REVIEW

File: srcpkgs/kconfigwidgets/template [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'kconfigwidgets'
+pkgname=kconfigwidgets
+version=5.26.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
+hostmakedepends="extra-cmake-modules"
+makedepends="qt5-devel kauth-devel kcodecs-devel kconfig-devel kguiaddons-devel kdoctools ki18n-devel kwidgetsaddons-devel"
```
Comment (Vaelatern) [inline srcpkgs/kconfigwidgets/template]: Does `kdoctools` belong in `makedepends`?

File: srcpkgs/kdesignerplugin/template [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'kdesignerplugin'
+pkgname=kdesignerplugin
+version=5.26.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
+hostmakedepends="extracmake-modules"
+makedepends="kdoctools kplotting-devel kdewebkit-devel"
```
Comment (Vaelatern) [inline srcpkgs/kdesignerplugin/template]: You use `kdoctools` again here, I guess I should find a listing of the package and answer my own question

File: srcpkgs/kfilemetadata5/template [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'kfilemetadata5'
+pkgname=kfilemetadata5
+version=5.26.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
+wrksrc="${pkgname%5}-${version}"
+hostmakedepends="extra-cmake-modules"
+makedepends="karchive-devel ki18n-devel poppler-qt5-devel taglib-devel exiv2-devel ffmpeg-devel catdoc attr-devel ebook-tools-devel"
```
Comment (Vaelatern) [inline srcpkgs/kfilemetadata5/template]: I don't see what `catdoc` is doing here.

File: srcpkgs/kglobalaccel/template [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'kglobalaccel'
+pkgname=kglobalaccel
+version=5.26.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
+hostmakedepends="extra-cmake-modules pkg-config"
+makedepends="kservice-devel pkg-config"
```
Comment (Vaelatern) [inline srcpkgs/kglobalaccel/template]: `pkg-config` in `makedepends`?

File: srcpkgs/kinit/template [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'kinit'
+pkgname=kinit
+version=5.26.0
+revision=1
+build_style=cmake
+configure_args="-DBUILD_TESTING=OFF"
+hostmakedepends="extra-cmake-modules libcap-progs"
+makedepends="extra-cmake-modules kio-devel kdoctools libcap-devel"
```
Comment (Vaelatern) [inline srcpkgs/kinit/template]: `extra-cmake-modules` in `makedepends`?

**Comment by dutsky** [comment]
I hope 1st pass issues fixed

**Comment by ebfe** [comment]
This needs a rebase to get all the dependencies

**Comment by dutsky** [comment]
With qt5 cross-building fixed, I must do some work on this PR it seems.

**Comment by pullmoll** [comment]
@dutsky You don't have to do that before merge. We can fix packages one after another in a joint effort.

**Comment by dutsky** [comment]
Oh, okay :)

**Comment by dutsky** [comment]
just curious, must I do something else before you can merge it? Thanks

**Comment by Vaelatern** [comment]
I just got control of my build box back, so I will be building each package and just making sure it looks okay.

**Comment by Vaelatern** [comment]
```
masterdir/destdir/kdoctools-5.26.0/usr/lib
masterdir/destdir/kdoctools-5.26.0/usr/lib/libKF5XsltKde.a
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools/KF5DocToolsConfigVersion.cmake
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools/KF5DocToolsConfig.cmake
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools/KF5DocToolsTargets-release.cmake
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools/KF5DocToolsMacros.cmake
masterdir/destdir/kdoctools-5.26.0/usr/lib/cmake/KF5DocTools/KF5DocToolsTargets.cmake
```
Should those go in a `-devel` subpackage?

**Comment by dutsky** [comment]
I just though it always used as makedepends and hostmakedepends at the same time, so i decided do not split it. I may be wrong, so i'll split it then.

**Comment by Vaelatern** [comment]
All the rest looks good to me, but it's massive and I have no way to really test it. Get back to me on the kdoctools files, and I think this is just about merge ready.

**Comment by dutsky** [comment]
Ready, I hope

**Comment by Duncaen** [comment]
Some musl related patches, posted in #xbps.
```
awilfox | https://bugs.kde.org/show_bug.cgi?id=365923 https://bugs.kde.org/show_bug.cgi?id=365925 https://bugs.kde.org/show_bug.cgi?id=365917 
```

**Comment by awilfox** [comment]
Hi, I'm awilfox from IRC; it appears the ki18n patch (third link) and ksolid patch (first link) are already applied.

The second link is for KIO's Proxy Autoconfiguration service.  It doesn't compile on musl because either ```_BSD_SOURCE``` or ```_GNU_SOURCE``` needs to be defined first.  I suggested in the [KDE bug](https://bugs.kde.org/show_bug.cgi?id=365925) to modify ```src/kpac/ConfigureChecks.cmake```, but they haven't responded yet.  In the meantime, an easy fix is [this patch](https://code.foxkit.us/adelie/patches/raw/master/kde-frameworks/kio/kio-5.21.0-bsd_source-definition.patch) I wrote that simply adds ```#define _BSD_SOURCE``` to the top of ```src/kpac/kpac_dhcp_helper.c```.

Otherwise, these packages should work on x86_64-musl as far as I can see.

**Comment by dutsky** [comment]
Thank you @Duncaen and @awilfox!
I can confirm, that I've already applied @awilfox's patch to ki18n, and ksolid patch already applied by someone.
Unfortunately i can't test kio patch at this moment, because something broken in qt5 crossbuilding with gcc 6.2, or it's just for me, but I'll try again later

**Pull request merged**